From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 15 May 2021 20:04:09 +0200
Subject: [PATCH] {BUG in 5.13-rc1} drivers/media/v4l2-core: use memcmp to
 match \0
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

A gcc plugin reported:

    drivers/media/v4l2-core/v4l2-jpeg.c: In function ‘jpeg_parse_app14_data’:
    drivers/media/v4l2-core/v4l2-jpeg.c:462:6: error: Invalid call to
    ‘strncmp’: string with nul characters [-Werror]
      462 |      strncmp(stream->curr, "Adobe\0", 6))
          |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
---
 drivers/media/v4l2-core/v4l2-jpeg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/v4l2-core/v4l2-jpeg.c b/drivers/media/v4l2-core/v4l2-jpeg.c
index 94435a7b6816..9be8811bb9f0 100644
--- a/drivers/media/v4l2-core/v4l2-jpeg.c
+++ b/drivers/media/v4l2-core/v4l2-jpeg.c
@@ -459,7 +459,7 @@ static int jpeg_parse_app14_data(struct jpeg_stream *stream,
 
 	/* Check for "Adobe\0" in Ap1..6 */
 	if (stream->curr + 6 > stream->end ||
-	    strncmp(stream->curr, "Adobe\0", 6))
+	    memcmp(stream->curr, "Adobe\0", 6))
 		return jpeg_skip(stream, lp - 2);
 
 	/* get to Ap12 */
-- 
